packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -238,23 +249,29 @@ class ImagePickerPlugin extends ImagePickerPlatform {
   }
 
   /// Monitors an <input type="file"> and returns the selected file(s).
-  Future<List<XFile>> _getSelectedXFiles(html.FileUploadInputElement input) {
+  Future<List<XFile>> _getSelectedXFiles(html.FileUploadInputElement input,
+      {double? maxWidth, double? maxHeight, int? imageQuality}) {
     final Completer<List<XFile>> _completer = Completer<List<XFile>>();
     // Observe the input until we can return something
-    input.onChange.first.then((event) {
+    input.onChange.first.then((event) async {
       final files = _handleOnChangeEvent(event);
       if (!_completer.isCompleted && files != null) {
-        _completer.complete(files
-            .map((file) => XFile(
-                  html.Url.createObjectUrl(file),
-                  name: file.name,
-                  length: file.size,
-                  lastModified: DateTime.fromMillisecondsSinceEpoch(
-                    file.lastModified ?? DateTime.now().millisecondsSinceEpoch,
-                  ),
-                  mimeType: file.type,
-                ))
-            .toList());
+        //Convert the file  if one of the parameter is not null
+        if (imageQuality != null || maxWidth != null || maxHeight != null) {
+          final convertedFileUris = await Future.wait(files.map((e) async =>
+              await _getCompressedUri(e, imageQuality, maxHeight, maxWidth)));
```

Comment:
This `_getCompressedUri` method should be a method from a separate object that only deals with resizing images in `XFile` objects.